crates/csharp/src/function.rs

```diff
@@ -948,7 +935,7 @@ impl Bindgen for FunctionBindgen<'_, '_> {
                 uwrite!(
                     self.src,
                     "
-                    var {array} = new List<{ty}>({length});
+                    var {array} = new global::System.Collections.Generic.List<{ty}>({length});
```

Comment:
Does `type_name_with_qualifier` give `ty` the global name too ?

Reply:
I do not believe so. Please lead me in the right direction to make this happen. It should be unnecessary as long as `ty` is either a primitive or a type already contained within the same namespace as this emitted code.

Reply:
I don't know much about the generator codebase, @jsturtevant please advise.

I just noticed that different flavors `ty` is sprinkled thru the code.

I'm thinking that we could have built-in hash table (or switch) translating known system types from short name to long name.  All of them for consistency sake. 

For `int` -> `global::System.Int32`

I guess later we will also generate marshaling for `System.Threading.Tasks.Task` which is not in `System` namespace.

And I think WIT can do list of list ?
`global::System.Collections.Generic.List<global::System.Collections.Generic.List<global::System.Int32>>`

Reply:
I don't personally believe that primitives (`int`) should be fully specified, but that's up to you.

Reply:
In general, one requirement of generated code (at least for .NET) is to fulfill the lowest common denominator. In C# that might mean not using interpolated strings, file-scoped namespaces, or other more modern features, as the user may target a language version which does not support those features.

`componentize-dotnet` is basically hard locked to projects which target .NET 10 and up, so this would only be a problem if the user explicitly downgraded using the `LangVersion` MSBuild property. It's not a scenario I would worry too much about, but I thought I'd mention it.

Reply:
Let's not worry about below Net10

Reply:
> Fully qualified type names are safer, as they leave no ambiguity.

This is compelling to me for generated code.  So I am +1 to this change.